Application/src/visuals/Message.java

```diff
@@ -112,39 +147,57 @@ public void render(GraphicsContext gc){
         y1 += offset; // Sets an offset from the previous message.
 
         // Checks if we are supposed to be animating the message.
-        if(keepAnimating) {
+        if (keepAnimating) {
             //Sets the message text centered above the "dragon".
-            gc.fillText(this.name, x1+animationBounds, y1 + (this.class_size - 2)); // Message description.
+            gc.fillText(this.name, x1 + animationBounds, y1 + (this.class_size - 2)); // Message description.
 
             // Checks if up image is supposed to be shown and if we are sending a message and not a return.
             if (switchImage && fromNode < toNode) {
                 //sets the dimensions of the dragon according to the current class size.
                 gc.drawImage(dragonMessage, x1 + animationBounds,
-                        y1 + (this.class_size), class_size/messageScale, class_size/messageScale); //State Wings Up.
+                        y1 + (this.class_size), class_size / messageScale, class_size / messageScale); //State Wings Up.
                 switchImage = false;
             }
 
             // Checks if down image is supposed to be shown and if we are sending a message and not a return.
             else if (!switchImage && fromNode < toNode) {
                 //sets the dimensions of the dragon according to the current class size.
                 gc.drawImage(dragonMessage2, x1 + animationBounds,
-                        y1 + (this.class_size), class_size/messageScale, class_size/messageScale); //State Wings Down.
+                        y1 + (this.class_size), class_size / messageScale, class_size / messageScale); //State Wings Down.
+                switchImage = true;
+            }
+
+            // Handling the switch of the dragon's flying direction when visualising self calls
+            // Checks if up image is supposed to be shown and if the direction of flying dragon should be switched.
+            else if (switchDirection && switchImage) {
+                //sets the dimensions of the dragon according to the current class size.
+                gc.drawImage(dragonMessage, x1 + animationBounds,
+                        y1 + (this.class_size), class_size / messageScale, class_size / messageScale); //State Wings Down.
+                //TODO: uncomment the following line after integration with the trail feature
+                //this.trails.add(new Trail((x1)+ animationBounds, y1 +(this.class_size),class_size/trailScale, class_size/trailScale));
```

Comment:
This shouldnt be here right

Reply:
Theres is a new self calling case that i added 
It needs the trail as well for full integration
Would you like the commented line to be removed before merging and then immediately after merging? Using a new hotfix branch?

Reply:
Its for a feature thats not implemented yet so you shouldnt have code related to it in your pull request, its going to be fixed when the integration between this feature and the trail feature happens. And that is not yet